index.bs

```diff
@@ -630,7 +589,199 @@ self.onmessage = async function(e) {
+automatic closing of {{VideoFrame}} and {{AudioData}} objects once they
+are written to a {{VideoTrackSource}}.
+
+# Backwards compatibility with earlier proposals # {#backwards-compatibility]}
+
+This section is informative.
+
+Previous proposals for this interface had an API like this:
+
+<pre class="idl">
+[Exposed=Window,DedicatedWorker]
+interface MediaStreamTrackGenerator : MediaStreamTrack {
+    constructor(MediaStreamTrackGeneratorInit init);
+    attribute WritableStream writable;  // VideoFrame or AudioFrame
+};
+
+dictionary MediaStreamTrackGeneratorInit {
+  required DOMString kind;
+};
+</pre>
+This interface had the generator for the MediaStreamTrack being an instance of
+a MediaStreamTrack rather than containing one.
+
+This interface can be shimmed on top of VideoTrackSource like this:
+
+<pre>
+
+Javascript goes here when written
+
+</pre>
+
+The VideoTrackSource can be shimmed on top of MediaStreamTrackGenerator like this:
+
+<pre>
+
+More Javascript goes here
+
+</pre>
+
+# Considerations for sourcing audio tracks # {#audio-processor}
+
+These are kept in the document to allow further discussion of audio processing, if
+feasible. They do not indicate that the WG has consensus to add audio.
+
+The following constrainable properties were defined for audio {{MediaStreamTrack}}s
+sourced from a {{MediaStreamTrackGenerator}}, but in an informational capacity
+only available via {{MediaStreamTrack/getSettings}}. It is not possible to
+reconfigure an audio track sourced by a {{MediaStreamTrackGenerator}} using
+{{MediaStreamTrack/applyConstraints}} in the same way that it is possble to,
+for example, resize the frames of a video track.
+{{MediaStreamTrack/getCapabilities}} MUST return an empty object.
+<table>
+  <thead>
+    <tr>
+      <th>
+        Property Name
+      </th>
+      <th>
+        Values
+      </th>
+      <th>
+        Notes
+      </th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr id="def-constraint-sampleRate">
+      <td data-tests="">
+        <dfn>sampleRate</dfn>
+      </td>
+      <td>
+        {{ConstrainDouble}}
+      </td>
+      <td>
+        As a setting, this is the sample rate, in samples per second, of the
+        latest {{AudioData}} delivered by the track.
+      </td>
+    </tr>
+    <tr id="def-constraint-channelCount">
+      <td data-tests="">
+        <dfn>channelCount</dfn>
+      </td>
+      <td>
+        {{ConstrainULong}}
+      </td>
+      <td>
+        As a setting, this is the number of independent audio channels of the
+        latest {{AudioData}} delivered by the track.
+      </td>
+    </tr>
+    <tr id="def-constraint-sampleSize">
+      <td data-tests="">
+        <dfn>sampleSize</dfn>
+      </td>
+      <td>
+        {{ConstrainULong}}
+      </td>
+      <td>
+        As a setting, this is the linear sample size of the latest {{AudioData}}
+        delivered by the track.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Example using audio/video co-processing
```

Comment:
I think the two code examples dedicated to audio need to stay in alvestrand until we adopt an audio API.

Also, all examples are still main-thread based, and show the old API, which I think needs to be updated.

I provided 3 examples (3.1, 3.2. 3.3) in https://jan-ivar.github.io/mediacapture-transform/#examples I recommend we use to show how to transfer MST to a worker and using the API there.